src/CaptureClient/ApiEventIdProvider.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureClient/ApiEventIdProvider.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <memory>
+
+namespace orbit_capture_client {
+
+[[nodiscard]] std::unique_ptr<NameEqualityApiEventIdProvider>
+NameEqualityApiEventIdProvider::Create(const orbit_grpc_protos::CaptureOptions& capture_options) {
+  const auto& instrumented_functions = capture_options.instrumented_functions();
+  const uint64_t max_id =
+      instrumented_functions.empty()
+          ? 0
+          : std::max_element(
+                std::begin(instrumented_functions), std::end(instrumented_functions),
+                [](const auto& a, const auto& b) { return a.function_id() < b.function_id(); })
+                ->function_id();
+  return std::make_unique<NameEqualityApiEventIdProvider>(max_id + 1);
+}
+
+NameEqualityApiEventIdProvider::NameEqualityApiEventIdProvider(uint64_t start_id)
+    : next_id_(start_id) {}
+
+[[nodiscard]] uint64_t NameEqualityApiEventIdProvider::ProvideId(const TimerInfo& timer_info) {
+  const std::string& name = timer_info.api_scope_name();
```

Comment:
Of course `GitHub` makes it impossible to reply sometimes...

>> So if I have an ApiScope and an ApiScopeAsync with the same name, they would get the same id? This sounds like a bug.
>
> I thought, this is intended behaviour. @dpallotti , do you also think sync and async events of the same name should not be grouped?

I think I'm with @florian-kuebler here. The way "normal" scopes and async scopes work is very different, so merging them if they have the same name can be confusing. The fact that the grouping of the former is by the string on the slice and the grouping of the latter is by the string of the track is a big difference in itself. Also in the "Live" tab we might want to show the source and I think it would be useful to distinguish between the two types of slices.

Reply:
Agree. Especially on the Live tab part. It should be possible to see if the entry there is dynamic instrumentation, usual manual instrumentation or async manual instrumentation.

Reply:
I'll use the composite key of the type and the name.